src/lumigo_tracer/parsers/parser.py

```diff
@@ -94,13 +95,39 @@ def parse_response(self, url: str, status_code: int, headers, body: bytes) -> di
 class DynamoParser(ServerlessAWSParser):
     should_add_message_id = False
 
+    def _extract_message_id(self, body: dict, method: str):
+        if method == "PutItem" and body.get("Item"):
+            return md5hash(body["Item"])
+        elif method in ("UpdateItem", "DeleteItem") and body.get("Key"):
+            return md5hash(body["Key"])
+        elif method == "BatchWriteItem" and body.get("RequestItems"):
+            first_item = next(iter(body["RequestItems"].values()))
+            if first_item:
+                if first_item[0].get("PutRequest"):
+                    return md5hash(first_item[0]["PutRequest"]["Item"])
+                else:
+                    return md5hash(first_item[0]["DeleteRequest"]["Key"])
+
+    def _extract_table_name(self, body: dict, method: str):
+        name = body.get("TableName")
+        if not name and method == "BatchWriteItem" and isinstance(body.get("RequestItems"), dict):
+            return next(iter(body["RequestItems"]))
```

Comment:
If `body["RequestItems"] == {}` (empty dict) you'll get an exception:
![image](https://user-images.githubusercontent.com/49228804/87028279-13ae0800-c1e7-11ea-9b20-b47a001490f0.png)

Reply:
This is not a legal query - boto3 will return an error...